specs/core/0_beacon-chain.md

```diff
@@ -342,14 +345,23 @@ class HistoricalBatch(Container):
     state_roots: Vector[Root, SLOTS_PER_HISTORICAL_ROOT]
 ```
 
+#### `DepositMsg`
+
+```python
+class DepositMsg(Container):
```

Comment:
Should it be `DepositData` and `SignedDepositData`?

Reply:
Didn't want to change existing names too much. Note that genesis code already merkleizes deposit-data including signatures, and there are a lot of clients who call it "deposit_data" all over the place, to work-around collisions with "deposit". Calling the inner type the message condenses the amount of changes to the bare minimum: the signature step briefly uses the type, everything else stays as-is.

This brings me to another part of the reason: that `Data` already has meaning in respect to the `Deposit` that adds a merkle proof for the data. I like to keep names with only a single add-on like that to describe what is happening.
The more extreme option would be to do: `MerkledSignedDeposit`, `SignedDeposit`, `Deposit`. But it's probably way too late for that, and maybe too verbose.

And note that with blocks etc. I think it is better the other way around: the signature is a thing to be verified as one of the very first things when processing a block. After that it can be forgotten and ignored. All the existing code that is defined for a `BeaconBlock` type can stay as-is, allowing to process unsigned blocks for testing etc. (also hear complaints from Nimbus who previously had problems with empty/invalid placeholder signatures). And `SignedBeaconBlock` is only really used at the outer signature-checking layer, minimizing the amount of changes.

Reply:
Agreed that using the `Object`/`SignedObject` pattern in naming for Deposit should be avoided because it doesn't actually follow the pattern.